pkg/streamingpromql/planning.go

```diff
@@ -190,21 +190,10 @@ func (p *QueryPlanner) NewQueryPlan(ctx context.Context, qs string, timeRange ty
 		}
 
 		if p.enableDelayedNameRemoval {
```

Comment:
The introduction of `NameDrop` is a breaking change, and our first breaking change since we merged https://github.com/grafana/mimir/pull/12808.

So if delayed name removal is enabled and we're going to add a `NameDrop` node, we should bump the query plan version to at least v1, if it's not already higher. 

And we should set `MaximumSupportedQueryPlanVersion` to 1 in this PR as well.